File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第2章平方和.cpp
//平方和  2019年第十届蓝桥杯省赛C/C++大学A组
//lanqiaoOJ 599 

#include<bits/stdc++.h>
using namespace std;
bool check(int n){
    while(n){
        if(n%10==1||n%10==2||n%10==9||n%10==0)
            return true;
        n/=10;
    }
    return false;
}
int main(){
    long long sum=0;
    for(int i=1;i<=2019;i++)
        if(check(i))
            sum+=i*i;
    cout<<sum;
}

File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第2章平方和.py
#平方和  2019年第十届蓝桥杯省赛C/C++大学A组
#lanqiaoOJ 599

sum = 0
for i in range(1,2020):
    s = str(i)
    if '2' in s or '0' in s or '1' in s or '9' in s:   sum += i*i
print(sum)
